bin/sdk/gutenberg.js

```diff
@@ -20,6 +20,33 @@ const omitPlugins = [
 	webpack.HotModuleReplacementPlugin,
 ];
 
+const wordpressPackages = [
+	'api-fetch',
+	'block-serialization-spec-parser',
+	'blocks',
+	'components',
+	'data',
+	'editor',
+	'element',
+	'i18n',
+];
+
+const getWordPressExternals = () =>
+	wordpressPackages.reduce( ( externals, package ) => {
+		externals[ `@wordpress/${ package }` ] = {
+			window: [
```

Comment:
E.g.

```js
{
	`@wordpress/${ package }`: `wp.${ package }`
}
```

or

```js
{
	`@wordpress/${ package }`: [ 'wp', /* camelCased package */ ]
}
```

(both untested)

Reply:
Also, shame that `_.mapValues()` doesn't allow us to change key names :slightly_frowning_face:

Reply:
Well, the above isn't valid syntax - template strings are expressions, and expressions can't be used when defining property names statically - only string literals or identifiers can be used there. And lodash doesn't have a good method to help us here, because we're converting an array to object, and we're both transforming the result keys and values.

Optionally, we could use `_.transform()` and have something like this:

```
transform( wordpressPackages, ( externals, package ) => {
	externals[ `@wordpress/${ package }` ] = {
		window: [
			'wp',
			package.replace(
				/-([a-z])/g,
				( match, letter ) => letter.toUpperCase()
			)
		]
	};
}, {} );
```

but to me that's almost the same as using `reduce()`, but without using an extra lodash helper.

Also, I did try other ways to make it work with global variables, but it seems we specifically need to map it to `window` to make that work.

Let me know if you have any other suggestions, I'm happy to reconsider.

Reply:
> Well, the above isn't valid syntax - template strings are expressions, and expressions can't be used when defining property names statically - only string literals or identifiers can be used there.

Granted, my example was missing brackets as the required level of indirection here, but adding those should make it viable:

```js
{
	[ `@wordpress/${ package }` ]: `wp.${ package }`
}
```

Reply:
Could you please provide a more complete example? I'm not sure I understand how this is applicable in this particular use case.

Reply:
I was hoping we could make it work with [string]{https://webpack.js.org/configuration/externals/#string} or [array]{https://webpack.js.org/configuration/externals/#array} values, by literally using the examples I was giving as key/value pairs in `externals`. However, I couldn't get either to work, so let's proceed with yours.